modules/dcache/src/main/java/diskCacheV111/services/PoolStatisticsV0.java

```diff
@@ -345,19 +284,19 @@ public void getInfo(PrintWriter pw) {
     public void messageArrived(CellMessage message) {}
 
     private String getNiceDayOfCalendar(Calendar calendar) {
-        return _dayOfCalendar.get().format(calendar.getTime());
+        return _dayOfCalendar.format(calendar.getTime().toInstant());
```

Comment:
The tima API always requires the timezone. Thus you should use something like:
`Calendar.getInstance().toInstant().atZone(ZoneId.systemDefault()`

This is easy to test with `jshell` (a java on command line):

```
[nairi] ~ $ jshell
|  Welcome to JShell -- Version 11.0.12
|  For an introduction type: /help intro

jshell> import java.time.*

jshell> import java.time.format.*

jshell> var formatter = DateTimeFormatter.ofPattern("yyyy" + File.separator + "MM" + File.separator + "dd" + File.separator + "yyyy-MM-dd-HH'.raw'");
formatter ==> Value(YearOfEra,4,19,EXCEEDS_PAD)'/'Value(MonthOf ... -'Value(HourOfDay,2)'.raw'

jshell> formatter.format(Calendar.getInstance().toInstant());
|  Exception java.time.temporal.UnsupportedTemporalTypeException: Unsupported field: YearOfEra
|        at Instant.getLong (Instant.java:602)
|        at DateTimePrintContext.getValue (DateTimePrintContext.java:308)
|        at DateTimeFormatterBuilder$NumberPrinterParser.format (DateTimeFormatterBuilder.java:2702)
|        at DateTimeFormatterBuilder$CompositePrinterParser.format (DateTimeFormatterBuilder.java:2341)
|        at DateTimeFormatter.formatTo (DateTimeFormatter.java:1843)
|        at DateTimeFormatter.format (DateTimeFormatter.java:1817)
|        at (#4:1)

jshell> formatter.format(Calendar.getInstance().toInstant().atZone(ZoneId.systemDefault()));
$5 ==> "2021/09/02/2021-09-02-10.raw"

jshell> 
```

Reply:
Yes, you're right. I've already noticed this problem in IntelliJ. I fixed it by implementing a method, which converts the calendar datatype to LocalDateTime. LocalDateTime can be used by the formatter:

    private LocalDateTime toLocalDateTime (Calendar calendar) {
        return calendar.getTime().toInstant().atOffset(ZoneOffset.UTC).toLocalDateTime();
    }

    private String getNiceDayOfCalendar(Calendar calendar) {
        return _dayOfCalendar.format(toLocalDateTime(calendar));
    }

But your solution is probably the better one, isn't it?

Reply:
You should probably use the system default zone instead of UTC, at least.